procrastinate/tasks.py

```diff
@@ -81,7 +81,7 @@ def __init__(
         self,
         func: Callable,
         *,
-        app: app.App,
+        app: Optional[app.App],
```

Comment:
@ewjoachim this is needed to avoid `mypy` complaints when a Blueprint creates a task with `app = None`.

I'm not 100% sure if there are any 2nd/3rd order implications from this, hence bringing it to your attention.

All tests pass and mypy has now stopped complaining about it :man_shrugging: